x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java

```diff
@@ -301,15 +284,9 @@ public void testCloseUnassignedFailedJobAndStopUnassignedStoppingDatafeed() thro
     public void testStopAndForceStopDatafeed() throws Exception {
         internalCluster().ensureAtMostNumDataNodes(0);
         logger.info("Starting dedicated master node...");
-        internalCluster().startNode(Settings.builder()
-            .put("node.master", true)
-            .put("node.data", true)
-            .put("node.ml", false)
-            .build());
+        internalCluster().startMasterOnlyNode();
         logger.info("Starting ml and data node...");
-        internalCluster().startNode(Settings.builder()
-            .put("node.master", false)
-            .build());
+        internalCluster().startNode(onlyRoles(Set.of(DiscoveryNodeRole.DATA_ROLE, DiscoveryNodeRole.MASTER_ROLE)));
```

Comment:
The code here used to disable master, but now it is enabled. Which is correct?

Reply:
Oops, that should have been `MachineLearning.ML_ROLE`.